src/plugins/navigation/public/plugin.tsx

```diff
@@ -129,39 +127,65 @@ export class NavigationPublicPlugin
 
     const config = this.initializerContext.config.get();
     const {
-      solutionNavigation: { featureOn: isSolutionNavigationFeatureOn },
+      solutionNavigation: { defaultSolution },
     } = config;
 
     const onCloud = cloud !== undefined; // The new side nav will initially only be available to cloud users
     const isServerless = this.initializerContext.env.packageInfo.buildFlavor === 'serverless';
-    const isSolutionNavEnabled = isSolutionNavigationFeatureOn && onCloud && !isServerless;
-    this.isSolutionNavEnabled$ = of(isSolutionNavEnabled);
 
-    if (isSolutionNavEnabled) {
-      chrome.project.setCloudUrls(cloud);
-      this.addDefaultSolutionNavigation({ chrome });
+    let isFeatureEnabled$ = of(false);
+    this.isSolutionNavEnabled$ = of(false);
 
-      this.isSolutionNavEnabled$ = combineLatest([
-        core.settings.globalClient.get$<boolean>(ENABLE_SOLUTION_NAV_UI_SETTING_ID),
-        core.settings.globalClient.get$<SolutionNavigationOptInStatus>(
-          OPT_IN_STATUS_SOLUTION_NAV_UI_SETTING_ID
-        ),
-        this.userProfileOptOut$,
-      ]).pipe(
-        takeUntil(this.stop$),
-        debounceTime(10),
-        map(([enabled, status, userOptedOut]) => {
-          if (!enabled || userOptedOut === true) return false;
-          if (status === 'hidden' && userOptedOut === undefined) return false;
-          return true;
-        })
+    if (cloudExperiments) {
+      isFeatureEnabled$ = from(
+        cloudExperiments.getVariation(SOLUTION_NAV_FEATURE_FLAG_NAME, false)
```

Comment:
This needs to be tested end-to-end with launchdarkly, this might not work because of the existing client-side race condition :( 
https://github.com/elastic/kibana/issues/167240


At this point there is a chance that launch darkly isn't fully loaded and it would just return the default value.

I think we should fix that bug first, I don't think we should try to hack the workaround on our side